test/CMakeLists.txt

```diff
@@ -55,22 +41,45 @@ function(pmpo_add_fortran_exe exename srcname)
   install(TARGETS ${exename} DESTINATION bin)
 endfunction()
 
+pmpo_add_exe(testWachspress testWachspress.cpp)
+pmpo_add_exe(unitTest unitTest.cpp)
+pmpo_add_exe(mpUnitTest mpUnitTest.cpp)
+pmpo_add_exe(testReconstruction testReconstruction.cpp)
+pmpo_add_exe(testTracking testTracking.cpp)
+pmpo_add_exe(testRebuild testRebuild.cpp)
+pmpo_add_exe(timeAssmblyWachspress testTiming.cpp)
 pmpo_add_fortran_exe(testFortranInit testFortranInit.f90)
-add_test(NAME testFortranInit COMMAND ./testFortranInit)
-
 pmpo_add_fortran_exe(testFortran testFortran.f90)
-add_test(NAME testFortran COMMAND ./testFortran)
+pmpo_add_fortran_exe(testFortranReadMPAS testFortranReadMPAS.f90)
 
-set(TEST_NC_FILE_PLANAR "${CMAKE_SOURCE_DIR}/test/sample_mpas_meshes/planar_nonuniform_cvt_for_square_737elms.nc")
+#add test
+function(pmpo_add_test testname command)
+  find_program(MPIRUN_EXECUTABLE NAMES mpirun)
+  if(MPIRUN_EXECUTABLE)
+    add_test(NAME ${testname} 
+             COMMAND mpirun --bind-to core -np 1 ${command} ${ARGN})
+  else()
+    add_test(NAME ${testname} COMMAND ${command} ${ARGN})
+  endif()
+endfunction()
```

Comment:
The `find_program(MPIRUN_EXECUTABLE NAMES mpirun)` call and associated logic should be moved outside the this function.  As is, IIUC, everytime we call `pmpo_add_test(...)` cmake will search for the `mpirun` binary.

The following *should* work, but I have not tested it.

```suggestion
find_program(MPIRUN_EXECUTABLE NAMES mpirun) 

function(pmpo_add_test testname command)
  if(MPIRUN_EXECUTABLE)
    add_test(NAME ${testname} 
             COMMAND mpirun --bind-to core -np 1 ${command} ${ARGN})
  else()
    add_test(NAME ${testname} COMMAND ${command} ${ARGN})
  endif()
endfunction()
```